src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java

```diff
@@ -177,37 +179,52 @@ protected void execute() {
       String dotNotatedIdColumnsString = util.listToRVector(dotNotatedIdColumns);
 
-      
+      // are we mbio stuffs or eigengene?
+      // presumably as we support more types in the future, this logic will become more complicated?
+      // might even involve subclassing plugins?
+      // i think we cross that bridge when we get there and know more.. 
+      // NOTE: getMember tells us the member type, rather than gives us a literal member
+      String collectionMemberType = collection.getMember() == null ? "unknown" : collection.getMember();
+      boolean isEigengene = false;
+      if (collectionMemberType.contains("Eigengene")) {
+        isEigengene = true;
+      }
       
       // Format inputs for R  
-      connection.voidEval("sampleMetadata <- SampleMetadata(data = sampleMetadata" +
-                                ", recordIdColumn=" + singleQuote(computeEntityIdColName) +
-                                ", ancestorIdColumns=as.character(" + dotNotatedIdColumnsString + "))");
-
-      // !!!!!!! TEMPORARY HACK !!!!!!!
-      // This to let the negative values seen in wgcna eigengenes play in mbio-land until we get a better solution
-      // see https://github.com/VEuPathDB/microbiomeComputations/issues/81
-      connection.voidEval("abundanceDataIdColNames <- names(abundanceData)[grepl('stable_id', names(abundanceData))]");
-      connection.voidEval("abundanceDataIds <- abundanceData[,abundanceDataIdColNames, with=FALSE]");
-      connection.voidEval("abundanceData <- plyr::numcolwise(veupathUtils::shiftToNonNeg)(abundanceData)");
-      connection.voidEval("abundanceData[,abundanceDataIdColNames] <- abundanceDataIds");
-
-      connection.voidEval("abundanceData <- AbundanceData(data=abundanceData" + 
-                                ", sampleMetadata=sampleMetadata" +
-                                ", recordIdColumn=" + singleQuote(computeEntityIdColName) +
-                                ", ancestorIdColumns=as.character(" + dotNotatedIdColumnsString + ")" +
-                                ", imputeZero=TRUE)");                          
-      
-      // Run correlation!
-      connection.voidEval("computeResult <- microbiomeComputations::correlation(data1=abundanceData" +
-                                                          ", method=" + singleQuote(method) +
-                                                          proportionNonZeroThresholdRParam +
-                                                          varianceThresholdRParam +
-                                                          stdDevThresholdRParam +
-                                                          ", verbose=TRUE)");
-
-
-      String statsCmd = "writeStatistics(computeResult, NULL, TRUE)";
+      if (isEigengene)  {
+        connection.voidEval("abundanceData <- abundanceData[order(" + computeEntityIdColName + ")]; " + 
+          "abundanceData <- abundanceData[, -as.character(" + dotNotatedIdColumnsString+ "), with=FALSE];" +
```

Comment:
can we change this to 'assayData'? In the assaySelf version this naming distinction made it really clear what should be what.